### PR TITLE
Remove Dataflow runner references in WordCount examples.

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @davorbonaci

**Comment by peihe** [comment]
R: @kennknowles 
R: - @davorbonaci

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java [hunk 1]
```diff
@@ -63,13 +62,18 @@ public static void main(String[] args) {
...
-    // CHANGE 1/3: Your project ID is required in order to run your pipeline on the Google Cloud.
-    options.setProject("SET_YOUR_PROJECT_ID_HERE");
-    // CHANGE 2/3: Your Google Cloud Storage path is required for staging local files.
-    options.setTempLocation("gs://SET_YOUR_BUCKET_NAME_HERE/AND_TEMP_DIRECTORY");
+    PipelineOptions options = PipelineOptionsFactory.create();
+
+    // In order to run your pipeline on the Google Cloud, you need to make following
+    // runner specific changes:
+
+    // CHANGE 1/4: Select a dataflow PipelineRunner.
```
Comment (kennknowles) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: I think we can change this just slightly to try to make it more runner-independent. Something like "Select a Beam runner" and give a couple examples.

Comment (peihe) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: Updated

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java [hunk 2]
```diff
@@ -63,13 +62,19 @@ public static void main(String[] args) {
...
-    DataflowPipelineOptions options = PipelineOptionsFactory.create()
-        .as(DataflowPipelineOptions.class);
-    options.setRunner(BlockingDataflowPipelineRunner.class);
-    // CHANGE 1/3: Your project ID is required in order to run your pipeline on the Google Cloud.
-    options.setProject("SET_YOUR_PROJECT_ID_HERE");
-    // CHANGE 2/3: Your Google Cloud Storage path is required for staging local files.
-    options.setTempLocation("gs://SET_YOUR_BUCKET_NAME_HERE/AND_TEMP_DIRECTORY");
+    PipelineOptions options = PipelineOptionsFactory.create();
+
+    // In order to run your pipeline on the Google Cloud, you need to make following
```
Comment (davorbonaci) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: rm "on the Google Cloud".

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java [hunk 3]
```diff
@@ -63,13 +62,19 @@ public static void main(String[] args) {
...
+    // runner specific changes:
+
+    // CHANGE 1/4: Select a Beam runner, such as BlockingDataflowPipelineRunner,
```
Comment (davorbonaci) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: Change 1/X: Select a Beam runner, such as Blocking..Runner or FlinkPipelineRunner.
Change 2/X: Specify runner-required options. For Blocking..Runner, set project and temp location as follows. For XXXRunner, ...

File: examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java [hunk 4]
```diff
@@ -39,16 +38,16 @@
...
-    options.setRunner(BlockingDataflowPipelineRunner.class);
-
-    // CHANGE 1 of 3: Your project ID is required in order to run your pipeline on the Google Cloud.
-    options.setProject("SET_YOUR_PROJECT_ID_HERE");
-
-    // CHANGE 2 of 3: Your Google Cloud Storage path is required for staging local files.
-    options.setStagingLocation("gs://SET_YOUR_BUCKET_NAME_HERE/AND_STAGING_DIRECTORY");
+    PipelineOptions options = PipelineOptionsFactory.create();
+    // In order to run your pipeline on the Google Cloud, you need to make following
+    // runner specific changes:
```
Comment (davorbonaci) [inline examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java]: As above.

File: examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java [hunk 5]
```diff
@@ -61,7 +61,7 @@ public static void main(String[] args) {
          .via((KV<String, Long> wordCount) -> wordCount.getKey() + ": " + wordCount.getValue())
          .withOutputType(TypeDescriptors.strings()))
 
-     // CHANGE 3 of 3: The Google Cloud Storage path is required for outputting the results to.
+     // CHANGE 4 of 4: The Google Cloud Storage path is required for outputting the results to.
```
Comment (davorbonaci) [inline examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java]: where is change 3?

Comment (peihe) [inline examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java]: fixed

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java (hunk 2 above)
Comment (peihe) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: done

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java (hunk 3 above)
Comment (peihe) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: done

File: examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java (hunk 4 above)
Comment (peihe) [inline examples/java8/src/main/java/org/apache/beam/examples/MinimalWordCountJava8.java]: done

**Comment by peihe** [comment]
PTAL

File: examples/java8/pom.xml [hunk 6]
```diff
@@ -75,6 +75,12 @@
             <goals><goal>analyze-only</goal></goals>
             <configuration>
               <failOnWarning>true</failOnWarning>
+              <ignoredUnusedDeclaredDependencies>
+                <!-- runners are used by ServiceLoader. -->
```
Comment (davorbonaci) [inline examples/java8/pom.xml]: Not sure I understand this.

For example, you might want to add dependencies in the `runtime` scope.

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java [hunk 7]
```diff
@@ -63,13 +62,21 @@ public static void main(String[] args) {
...
+    //
+    // CHANGE 1/3: Select a Beam runner, such as BlockingDataflowPipelineRunner
+    // or FlinkPipelineRunner.
+    // CHANGE 2/3: Specify runner-required options.
+    // For BlockingDataflowPipelineRunner, set project and temp location as follows:
+    //   DataflowPipelineOptions dataflowOptions = options.as(DataflowPipelineOptions.class);
+    //   dataflowOptions.setRunner(BlockingDataflowPipelineRunner.class);
+    //   dataflowOptions.setProject("SET_YOUR_PROJECT_ID_HERE");
+    //   dataflowOptions.setTempLocation("gs://SET_YOUR_BUCKET_NAME_HERE/AND_TEMP_DIRECTORY");
+    // For FlinkPipelineRunner, check {@code FlinkPipelineOptions} for other options.
```
Comment (davorbonaci) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: For FlinkPipelineRunner, set the runner as follows. See {@link } for more details.

File: examples/java8/pom.xml (hunk 6 above)
Comment (peihe) [inline examples/java8/pom.xml]: Done

File: examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java (hunk 7 above)
Comment (peihe) [inline examples/java/src/main/java/org/apache/beam/examples/MinimalWordCount.java]: Done

**Comment by peihe** [comment]
Need another merge after PR482 is in.

**Comment by peihe** [comment]
rebased, it will be ready for merge once tests pass.

**Comment by davorbonaci** [comment]
There are build breaks.

**Comment by peihe** [comment]
tests passed, it is ready to merge.

**Pull request closed without merging**